include/albatross/src/graph/minimum_spanning_tree.hpp

```diff
@@ -173,6 +173,92 @@ Graph<VertexType> maximum_spanning_tree(const Graph<VertexType> &graph) {
   return output;
 }
 
+// A helper struct to track if a closed loop would be formed by a proposed edge
+// during Kruskal's algorithm. This struct associates each node with a tree,
+// initialized to each node in a unique tree. When a new edge is added, the
+// trees associated with the edge's nodes are merged into one tree. If a
+// proposed edge already has both nodes in the same tree, a closed loop would be
+// formed, and the edge can be rejected.
+template <typename VertexType> struct KruskalNodeTracker {
```

Comment:
nit: I think our code style would use `class` here instead of `struct` since it's non-passive : https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes